group/views.py

```diff
@@ -182,3 +182,26 @@ def post(self, request):
                             status=status.HTTP_400_BAD_REQUEST)
         return Response({"error": "User is not a member of this group"},
                         status=status.HTTP_403_FORBIDDEN)
+
+    def get(self, request, user_id):
+        """Retrieve all messages in a group"""
+        data = request.data
+        group_id = data.get("group_id")
+        try:
+            group = Group.objects.get(id=group_id)
+            user = User.objects.get(id=user_id)
```

Comment:
Use `select_related` here. That should also take care of the if condition on line 201.